Hyper/PlayerData/InventorySystem/Items/Item.cs

```diff
@@ -20,4 +20,28 @@ internal abstract class Item
     public virtual void Up() { }
 
     public virtual void Down() { }
+
+    public bool Equals(Item? other)
+    {
+        if (other is null)
+            return false;
+
+        if (ReferenceEquals(this, other))
+            return true;
+
+        if (GetType() != other.GetType())
+            return false;
+
+        return other.Id == Id;
+    }
+
+    public override bool Equals(object? obj)
+    {
+        return Equals(obj as Item);
+    }
+
+    public override int GetHashCode()
```

Comment:
You sure 2 items of the same kind should not have different hash codes?

Reply:
I think that's fine, items of the same kind are not discernable (at least that's how I look at it)